File: projects/lostengine/lost/rg/Draw.cpp
#include "lost/rg/Draw.h"
#include "lost/mesh/Mesh.h"
#include "lost/gl/Context.h"

namespace lost
{
namespace rg
{

Draw::Draw(mesh::MeshPtr inMesh)
: mesh(inMesh)
{
  name = "draw";
}

NodePtr Draw::create(mesh::MeshPtr inMesh)
{
  return DrawPtr(new Draw(inMesh));
}

void Draw::process(gl::ContextPtr ctx)
{
  if(!active) return;
  ctx->draw(mesh);
}

}
}

File: projects/lostengine/lost/lua/lost/io/rg/Draw.lua
-- lost.io.rg.Draw
module("lost.io", package.seeall)


require("lost.io.Loader")

-- mesh = (any meshPtr, will be cast to MeshPtr) (mandatory)
-- material
function Loader:Draw(definition)
  local result = lost.rg.Draw.create(definition["mesh"])
  
  if definition["modelTransform"] then
    result.modelTransform = definition["modelTransform"]
  end

  if definition["name"] then
    result.name = definition["name"]
  end

  if definition["active"] ~= nil then
    result.active = definition["active"]
  end		
  
  return result
end